File: crates/swiftlet/examples/gr_regex_flags.rs
use std::sync::Arc;
use swiftlet::{ParserConfig, Swiftlet};

fn main() {
    let grammar = r#"
    start: "hello"i
    "#;

    let texts = ["hello", "Hello", "HELLO", "HeLLo"];

    let conf = Arc::new(ParserConfig::default());
    let parser = Swiftlet::from_str(grammar)
        .map(|grammar| grammar.parser(conf))
        .expect("failed to build parser");
    for text in texts {
        match parser.parse(&text) {
            Ok(ast) => {
                ast.print()
                // Output:
                // Tree("start", ["hello"])
                // Tree("start", ["Hello"])
                // Tree("start", ["HELLO"])
                // Tree("start", ["HeLLo"])
            }
            Err(err) => eprintln!("{}", err),
        }
    }
}

File: pyswiftlet/examples/gr_regex_flags.py
"""Grammar regex flags example using the Swiftlet Python bindings."""

from swiftlet import Swiftlet


GRAMMAR = r"""
start: "hello"i
"""


def main() -> None:
    parser = Swiftlet(GRAMMAR)
    for text in ("hello", "Hello", "HELLO", "HeLLo"):
        tree = parser.parse(text)
        tree.pretty_print()


if __name__ == "__main__":
    main()
